BUG: _ComparisonOp doesn't support numeric interoperability?

### Describe the issue:

This (sort of) came up in https://github.com/python/mypy/issues/3186#issuecomment-1176593668 recently, but `_ComparisonOp`'s current form likely frustrates numeric interoperability. From [here](https://github.com/thai-phi/numpy_backup/blob/5d0517163889cb9519c3e2731b3737276d27cfa7/numpy/typing/_callable.py#L323-L327):

``` python
    class _ComparisonOp(Protocol[_T1, _T2]):
        @overload
        def __call__(self, __other: _T1) -> bool_: ...
        @overload
        def __call__(self, __other: _T2) -> _ArrayOrScalar[bool_]: ...
```

This may be well beyond the scope of what is possible with NumPy or Mypy (in which case, this should probably be closed as WONTFIX), but it does present a challenge. As long as one confines oneself to scalars on either side of the comparison, I'm pretty sure `numpy` will never return an array. I don't think that notion is sufficiently captured by the above.

### Reproduce the code example:

```python
# test_comparison.py
import numpy as np
from fractions import Fraction

assert np.int64(0) < Fraction(1) is True  # type error, but runs okay
```


### Error message:

```shell
test_comparison.py:5: error: No overload variant of "__call__" of "_ComparisonOp" matches argument type "Fraction"  [call-overload]
test_comparison.py:5: note: Possible overload variants:
test_comparison.py:5: note:     def __call__(self, Union[int, float, complex, number[Any], bool_]) -> bool_
test_comparison.py:5: note:     def __call__(self, Union[_SupportsArray[dtype[Union[bool_, number[Any]]]], _NestedSequence[_SupportsArray[dtype[Union[bool_, number[Any]]]]], bool, int, float, complex, _NestedSequence[Union[bool, int, float, complex]]]) -> ndarray[Any, dtype[bool_]]
Found 1 error in 1 file (checked 1 source file)
```


### NumPy/Python version information:

1.23.0 3.10.5 (main, Jun  7 2022, 08:51:33) [Clang 13.0.0 (clang-1300.0.29.30)]

--- a/numpy/_typing/_callable.pyi
+++ b/numpy/_typing/_callable.pyi
@@ -44,6 +44,7 @@ from ._scalars import (
 )
 from . import NBitBase
 from ._generic_alias import NDArray
+from ._nested_sequence import _NestedSequence
 
 _T1 = TypeVar("_T1")
 _T2 = TypeVar("_T2")
@@ -318,8 +319,20 @@ class _ComplexOp(Protocol[_NBit1]):
 class _NumberOp(Protocol):
     def __call__(self, other: _NumberLike_co, /) -> Any: ...
 
+class _SupportsLT(Protocol):
+    def __lt__(self, other: Any, /) -> object: ...
+
+class _SupportsGT(Protocol):
+    def __gt__(self, other: Any, /) -> object: ...
+
 class _ComparisonOp(Protocol[_T1_contra, _T2_contra]):
     @overload
     def __call__(self, other: _T1_contra, /) -> bool_: ...
     @overload
     def __call__(self, other: _T2_contra, /) -> NDArray[bool_]: ...
+    @overload
+    def __call__(
+        self,
+        other: _SupportsLT | _SupportsGT | _NestedSequence[_SupportsLT | _SupportsGT],
+        /,
+    ) -> Any: ...

--- a/numpy/typing/tests/data/reveal/comparisons.pyi
+++ b/numpy/typing/tests/data/reveal/comparisons.pyi
@@ -1,4 +1,6 @@
 import numpy as np
+import fractions
+import decimal
 
 c16 = np.complex128()
 f8 = np.float64()
@@ -24,6 +26,13 @@ AR = np.array([0], dtype=np.int64)
 AR.setflags(write=False)
 
 SEQ = (0, 1, 2, 3, 4)
+
+# object-like comparisons
+
+reveal_type(i8 > fractions.Fraction(1, 5))  # E: Any
+reveal_type(i8 > [fractions.Fraction(1, 5)])  # E: Any
+reveal_type(i8 > decimal.Decimal("1.5"))  # E: Any
+reveal_type(i8 > [decimal.Decimal("1.5")])  # E: Any
 
 # Time structures
 

